src/core/server/http/router/headers.ts

```diff
@@ -37,17 +37,15 @@ export type KnownHeaders = KnownKeys<IncomingHttpHeaders>;
  * Http request headers to read.
  * @public
  */
-export type Headers = { [header in KnownHeaders]?: string | string[] | undefined } & {
+export interface Headers {
   [header: string]: string | string[] | undefined;
-};
+}
 
 /**
  * Http response headers to set.
  * @public
  */
-export type ResponseHeaders =
-  | Record<KnownHeaders, string | string[]>
```

Comment:
`KnownHeaders` helper is broken, I didn't manage to fix it quickly. Its only purpose is to improve autocomplete in IDE, so I removed it for time being. I will fix it later.

Reply:
Ran into the same issue in Fleet. Just replacing with a `string[]` for now. 

https://github.com/elastic/kibana/blob/3a434d710e14dcdac15f5b78cfac48d887502e60/x-pack/plugins/fleet/server/routes/epm/handlers.ts#L164